### ISSUE TITLE
The type 'AddEventListenerOptions' is not defined

### ISSUE THREAD

**Issue by AlbertoDePena** [Zaid-Ajaj/Feliz#252]
Getting this error when running: `npm run start`

```bash
ERROR in ./.fable/Feliz.UseListener.0.6.0/Listener.fs
Module Error (from ./node_modules/fable-loader/index.js):
C:/Dev/ElmishProjects/text-classification/.fable/Feliz.UseListener.0.6.0/Listener.fs(99,88): (99,111) error FSHARP: The type 'AddEventListenerOptions' is not defined. (code 39)
 @ ./.fable/Feliz.Router.3.1.0/Router.fs 8:0-184 167:13-45 170:13-51 199:13-45 202:13-51 232:11-43 235:11-49
 @ ./src/App.fs
 @ ./src/Main.fs
 @ ./src/App.fsproj
```

```fsproj
 <ItemGroup>
    <PackageReference Include="Fable.Elmish.React" Version="3.0.1" />
    <PackageReference Include="Feliz" Version="1.13.0" />
    <PackageReference Include="Feliz.Router" Version="3.1.0" />
  </ItemGroup>
```

```json
{
  "private": true,
  "scripts": {
    "start": "webpack-dev-server",
    "build": "webpack --mode production"
  },
  "dependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1"
  },
  "devDependencies": {
    "@babel/core": "^7.11.4",
    "fable-compiler": "^2.11.0",
    "fable-loader": "^2.1.9",
    "webpack": "^4.44.1",
    "webpack-cli": "^3.3.12",
    "webpack-dev-server": "^3.11.0"
  }
}
```

```F#
// App.fs

[<RequireQualifiedAccess>]
module App

open Elmish
open Feliz
open Feliz.Router

type State =
  { CurrentUrl : string list }

type Msg =
  | UrlChanged of string list

let init () =
  { CurrentUrl = Router.currentUrl() }, Cmd.none

let update (msg: Msg) (state: State) =
  match msg with
  | UrlChanged segments -> 
    { state with CurrentUrl = segments }, Cmd.none

let render (state: State) (dispatch: Msg -> unit) =
  React.router [
    router.onUrlChanged (UrlChanged >> dispatch)
    router.children [
      match state.CurrentUrl with
      | [ ] -> Html.h1 "Classification Page"
      | [ "labels" ] -> Html.h1 "Labels page"
      | [ "text-samples" ] -> Html.h1 "Text Samples page"
      | _ -> Html.h1 "Not found"
    ]
  ]
```

```F#
// Main.fs

module Main

open Elmish
open Elmish.React

Program.mkProgram App.init App.update App.render
|> Program.withConsoleTrace
|> Program.withReactSynchronous "elmish-app"
|> Program.run
```

**Comment by Zaid-Ajaj** [comment]
Hi @AlbertoDePena, can you please try install latest `Fable.Browser.Event` package explicitly into the project to fix this?
```xml
<PackageReference Include="Fable.Browser.Event" Version="1.2.1" />
```

CC @MangelMaxime I think Fable.React needs to update to latest `Fable.Browser.Dom` because this problem is creeping up all the way up to Feliz :/

**Comment by MangelMaxime** [comment]
The error seems different than the overload one we saw before. By default, dotnet restore/install should restore the latest version of the transitive dependencies or the version that has been locked by a dependency. And TBH I don't think anyone locked the version to one of the versions that cause problems.

This time it complains about not finding the type `AddEventListenerOptions` which has not been changed in the last PRs.

It could be useful to have a reproduction repository for the error in this issue before rolling release for all the package which depends on Fable.Browser.Dom/Event.

@AlbertoDePena Would you please be able to provide a reproduction repository for your problem? Did you start from a template?

**Comment by AlbertoDePena** [comment]
@Zaid-Ajaj Adding `Fable.Browser.Event` did the trick...

@MangelMaxime  here is the repo I am using [text-classification (elmish branch)](https://github.com/AlbertoDePena/text-classification/tree/elmish)

**Comment by MangelMaxime** [comment]
I spoke too quickly `AddEventListenerOptions` is a rename from `EventTargetOptions`

The problem is kind of coming from `Feliz.useListener` because @Shmew forgot to add `Fable.Browser.Dom` to it's `paket.references` in it's commit [Update Fable.Browser.Dom dependency to fix breaking change](https://github.com/Shmew/Feliz.UseListener/commit/d2df2eb8e83a6e9b097fa2f1eabc026b00bff596#diff-b27553841b4413a24ed1b9aff865105bR10) so the NuGet package doesn't include the restriction on the API and use the one provided by resolving Feliz -> Fable.React -> Fable.Dom.Browser.

And because he is using the new Fable.Browser.Dom without forcing the "right" version of Fable.Browser.Dom then depending on how the user project is setup we can have a problem.

Anyway, I released Fable.React 7.0.1 hoping that it will make it easier to lock the "right" version of Fable.Browser.Dom.

-----

To summarize:

If Feliz force to require Fable.React >= 7.0.1 then @Shmew should just need to upgrade to latest version of Feliz and doesn't need to specify a version for Fable.Browser.Dom.

**Comment by Shmew** [comment]
Ah yeah I didn't think about that. 😓 

I'll update my library once we have Feliz targeting Fable.React >= 7.0.1 and put in a PR to Feliz.Router.

Thanks for the quick response @MangelMaxime! ❤️
